docs/source/style_guide.md

```diff
@@ -0,0 +1,187 @@
+# Documentation style guide
```

Comment:
Could this title be `Documentation Style guide for contributors`, with contributions also mentioned in the first paragraph. This could be a good opportunity to make the project more inviting.

Reply:
I worry about the name being too long, but it's worth trying.